Automattic/jetpack
# Make premium-content buttons work in notifications

Fixes #23696

#### Changes proposed in this Pull Request:

The subscribe button used in premium content blocks should open a dialog
to pay for the premium content when clicked, however when the subscribe
button was displayed in emails or in wp.com notifications it didn't do
that. It did work as intended when a recurring payment button was used
directly.

This was happening because the underlying jetpack/button wasn't using
the correct `url` - none was set.

This has been addressed by:
 1. Changing the template in premium-content to specify the
    `passThroughAttributes` on the jetpack/button, as it is in recurring-payments
 2. Updating recurring-payments so that if the `planId` prop is changed,
    it sets the `url` and `uniqueId` props too. Previously it only updated those 
    props if using the control from this block to set the payment plan

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Create a wpcom site with a paid plan
* With a separate user, subscribe to email and notifications for that site
* Create a post with a premium content block
* As the subscribed user click on the 'Subscribe' button in the email, it should open the post with the checkout form popover opened
* As the subscribed user click on the 'Subscribe' button in the notification, it should open the post with the checkout form popover opened

![image](https://user-images.githubusercontent.com/93301/163435618-6eb04ee6-3878-46f2-ab53-89188eabd208.png)
